use-cases/employee-scheduling/src/main/java/org/acme/employeescheduling/domain/Availability.java

```diff
@@ -69,4 +62,16 @@ public void setAvailabilityType(AvailabilityType availabilityType) {
     public String toString() {
         return availabilityType + "(" + employee + ", " + date + ")";
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof Availability that)) return false;
+        return Objects.equals(getId(), that.getId());
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getId());
```

Comment:
Let's not use [Objects.hash(...)](https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#hash-java.lang.Object...-).

This method creates a varargs array every time it is invoked. When this `hashCode` method gets on a hot path (such as in constraints), this is a considerable slowdown.

Let's not teach people this.